Update resolver package

folarinemurgo opened the pull request:
## Description / Change(s) / Related issue(s)
Sync impl. for csl init in test files.
Removed @tanstack/react-query and other depended on files, replaced with try and catch

##### Ticket
YV-465


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Refactor**
  * Removed all dependencies on `@tanstack/react-query` from the resolver package, switching to React's built-in state and effect hooks for data management.
  * Updated several custom hooks to use synchronous functions and direct state management.
  * Simplified test setups by removing query client fixtures and related logic.
  * Cleaned up configuration and removed unused dependencies.
  * Modified internal components to manage query clients internally rather than via props.

* **Bug Fixes**
  * Improved performance and reliability by converting certain asynchronous utility functions to synchronous execution.

* **Chores**
  * Removed unused files and configuration related to query client and mutation utilities.
  * Updated package dependencies and configuration for cleaner project setup.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Opestein committed:
rn79: Update: resolver package
Removed @tanstack/react-query and other depended on files, replaced with try and catch

Opestein committed:
rn79: Update: resolver package
Removed @tanstack/react-query and other depended on files, replaced with try and catch

Opestein committed:
rn79: Update: resolver package
Removed @tanstack/react-query and other depended on files, replaced with try and catch

Opestein committed:
rn79: Update: resolver package
Removed @tanstack/react-query and other depended on files, replaced with try and catch

Opestein committed:
rn79: Update: resolver package
Removed @tanstack/react-query and other depended on files, replaced with try and catch

Opestein committed:
rn79: Update: resolver package
Removed @tanstack/react-query and other depended on files, replaced with try and catch

Opestein committed:
rn79: Update: resolver package
Added cleanup to prevent memory leaks and verify dependency array

Opestein committed:
rn79: Update: resolver package
Added cleanup to prevent memory leaks and verify dependency array

folarinemurgo commented:
Please any update on PR approval? @stackchain

Opestein committed:
rn79: Update Resolver Package
Added more test cases; Statements and Lines => 100%

Opestein committed:
rn79: Update Resolver Package
Added more test cases; Statements and Lines => 100%

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverCryptoAddresses.tsx:
resolve is still async cuz it consumes some external APIs, also the {signal} is used to cancel the requests when the screen is unmounting to stop resolving it, these changes need to be reverted, this will continue to use react-query

folarinemurgo replied:
Guess i can close the PR then since it's about migration away from @tanstack/react-query?

Opestein committed:
rn79: Migrate-UI-Menu
Reverted react query for resolveruseResolverCryptoAddresses

Opestein committed:
rn79: Migrate-UI-Menu
Reverted react query for resolveruseResolverCryptoAddresses

Opestein committed:
Merge branch 'refs/heads/develop' into Update-resolver-package

stackchain commented on packages/resolver/src/adapters/cns/api-helpers.ts:
```diff
@@ -74,9 +74,9 @@ export const resolveUserRecord = async (
   const virtualSubdomains = await parseAssocMapAsync(
```
just a question, is it sill async? parseAssocMap?

folarinemurgo replied:
Yes, it's still async

stackchain commented on packages/resolver/package.json:
```diff
@@ -130,8 +129,9 @@
-    "@emurgo/cross-csl-nodejs":"8.0.0-alpha.1",
+    "@emurgo/cross-csl-nodejs": "8.0.0-alpha.1",
     "@react-native-async-storage/async-storage": "2.1.2",
+    "@tanstack/react-query": "^5.84.2",
```
@folarinemurgo just to check is it matching the other package versions?

folarinemurgo replied:
Updated to 5.76.2 (from other packages)

Opestein committed:
rn79: Migrate-UI-Menu
Updated @tanstack/react-query to 5.76.2 to match other packages

Opestein committed:
rn79: Migrate-UI-Menu
Updated @tanstack/react-query to 5.76.2 to match other packages

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverCryptoAddresses.tsx:
```diff
@@ -1,39 +1,29 @@
   return {
-    ...query,
-    cryptoAddresses: query.data ?? [],
+    data: data ?? [],
```
it returns the query this breaks the API, please return ...query, cuz other properties such as `onSettled` can be in use by the clients.

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverSetShowNotice.tsx:
```diff
@@ -1,21 +1,29 @@
+  const [isLoading, setIsLoading] = useState(false)
+  const [error, setError] = useState<Error | null>(null)
+
+  const setShowNotice = async (value: boolean) => {
```
useMutationWithInv... can be consumed from `@yoroi/common`, update the return to expand the mutation otherwise it will be a breaking.

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverShowNotice.tsx:
```diff
@@ -1,22 +1,45 @@
-    queryKey: ['useResolverShowNotice'],
-    queryFn: showNotice.read,
-  })
+  useEffect(() => {
```
same.

stackchain replied:
@folarinemurgo you can keep the mutations as well for now, we can migrate it later

Opestein committed:
rn79: Update Resolver Package
Replace manual useState/useEffect patterns with @tanstack/react-query in
useResolverSetShowNotice and useResolverShowNotice hooks for better
state management, caching, and error handling.

Opestein committed:
rn79: Update Resolver Package
Replace manual useState/useEffect patterns with @tanstack/react-query in
useResolverSetShowNotice and useResolverShowNotice hooks for better
state management, caching, and error handling.

Opestein committed:
Merge remote-tracking branch 'origin/develop' into Update-resolver-package

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverCryptoAddresses.tsx:
```diff
@@ -1,35 +1,22 @@
-    ...options,
-    queryFn: ({signal}) =>
-      crypto.getCardanoAddresses({resolve, strategy}, {signal}),
+    queryKey: ['resolver', 'crypto-addresses', resolve, strategy],
```
resolver MUST avoid caching.

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverSetShowNotice.tsx:
```diff
@@ -1,21 +1,18 @@
     ...mutation,
     setShowNotice: mutation.mutate,
-  }
+  } as any
```
missing options

stackchain replied:
import from `@yoroi/common` to invalidate the key immediately

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverShowNotice.tsx:
```diff
@@ -1,22 +1,19 @@
-    ...options,
-    queryKey: ['useResolverShowNotice'],
-    queryFn: showNotice.read,
+    queryKey: ['resolver', 'show-notice'],
```
missing options

Opestein committed:
rn79: Migrate-UI-Menu
removed caching and add options support to show notice hooks

Opestein committed:
rn79: Migrate-UI-Menu
removed caching and add options support to show notice hooks

Opestein committed:
Merge remote-tracking branch 'origin/develop' into Update-resolver-package

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverCryptoAddresses.tsx:
```diff
@@ -1,35 +1,24 @@
+}: {
+  resolve: Resolver.Receiver['resolve']
+  strategy?: Resolver.Strategy
+}) => {
```
@folarinemurgo the options is need the hooks and the cb are set in it, as pointed by cursor bot.
```
,
  options?: UseQueryOptions<
    Resolver.AddressesResponse,
    Error,
    Resolver.AddressesResponse,
    ['useResolverCryptoAddresses', string]
  >,
```

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverShowNotice.tsx:
```diff
@@ -1,22 +1,24 @@
+      return await showNotice.read()
+    },
+    staleTime: 0,
+    gcTime: 0,
```
that is also correct, it can also receive just the function like before.

stackchain reviewed (changes requested):
tiny changes to merge

Opestein committed:
Merge remote-tracking branch 'origin/develop' into Update-resolver-package

Opestein committed:
rn79: Update Resolver Package
enable useMutationWithInvalidations tests with proper Jest config

Opestein committed:
rn79: Update Resolver Package
enable useMutationWithInvalidations tests with proper Jest config

folarinemurgo commented:
> tiny changes to merge

All resolved

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverCryptoAddresses.tsx:
```diff
@@ -23,13 +21,13 @@ export const useResolverCryptoAddresses = (
   const query = useQuery({
-    throwOnError: true,
     queryKey: ['useResolverCryptoAddresses', resolve],
+    queryFn: async ({signal}: {signal?: AbortSignal}) => {
```
signal in this case is mandatory, cuz when the debounce of typing kicking again, it should cancel the previous request
```suggestion
    queryFn: async ({signal}: {signal: AbortSignal}) => {
```

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverSetShowNotice.tsx:
```diff
@@ -1,21 +1,24 @@
+    mutationFn: async (value: boolean) => {
+      await showNotice.save(value)
+    },
+    invalidateQueries: [['resolver', 'show-notice']],
```
some hooks we use the opposite hook
```suggestion
    invalidateQueries: [[useResolverShowNotice']],
```

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverSetShowNotice.tsx:
```diff
@@ -1,21 +1,24 @@
   const {showNotice} = useResolver()
 
   const mutation = useMutationWithInvalidations({
+    mutationFn: async (value: boolean) => {
+      await showNotice.save(value)
+    },
```
```suggestion
    mutationFn: showNotice.save,
```

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverSetShowNotice.tsx:
```diff
@@ -1,21 +1,24 @@
+      await showNotice.save(value)
+    },
+    invalidateQueries: [['resolver', 'show-notice']],
     ...options,
```
```suggestion
    ...options,
    mutationKey: [['useResolverShowNotice']],
```

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverSetShowNotice.tsx:
```diff
@@ -1,21 +1,24 @@
     ...mutation,
     setShowNotice: mutation.mutate,
-  }
+  } as any
```
```suggestion
  }
```

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverShowNotice.tsx:
```diff
@@ -1,22 +1,26 @@
   const {showNotice} = useResolver()
 
   const query = useQuery({
+    queryKey: ['resolver', 'show-notice'],
```
```suggestion
    queryKey: ['useResolverShowNotifce'],
```

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverShowNotice.tsx:
```diff
@@ -1,22 +1,26 @@
+    queryKey: ['resolver', 'show-notice'],
+    queryFn: async () => {
+      return await showNotice.read()
+    },
```
```suggestion
    queryFn: showNotice.read,
```

stackchain committed:
Update packages/resolver/src/translators/reactjs/hooks/useResolverSetShowNotice.tsx

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>

stackchain committed:
Update packages/resolver/src/translators/reactjs/hooks/useResolverSetShowNotice.tsx

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>

stackchain committed:
Update packages/resolver/src/translators/reactjs/hooks/useResolverShowNotice.tsx

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>

stackchain committed:
Update packages/resolver/src/translators/reactjs/hooks/useResolverShowNotice.tsx

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>

stackchain committed:
Update packages/resolver/src/translators/reactjs/hooks/useResolverShowNotice.tsx

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>

stackchain committed:
Update packages/resolver/src/translators/reactjs/hooks/useResolverShowNotice.tsx

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>

stackchain committed:
Update packages/resolver/src/translators/reactjs/hooks/useResolverSetShowNotice.tsx

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>

stackchain committed:
Update packages/resolver/src/translators/reactjs/hooks/useResolverSetShowNotice.tsx

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>

stackchain committed:
Update packages/resolver/src/translators/reactjs/hooks/useResolverSetShowNotice.tsx

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>

stackchain committed:
Update packages/resolver/src/translators/reactjs/hooks/useResolverSetShowNotice.tsx

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>

stackchain committed:
Update packages/resolver/src/translators/reactjs/hooks/useResolverSetShowNotice.tsx

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>

stackchain committed:
Update packages/resolver/src/translators/reactjs/hooks/useResolverSetShowNotice.tsx

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverShowNotice.tsx:
```diff
@@ -1,22 +1,24 @@
   const query = useQuery({
-    ...options,
-    queryKey: ['useResolverShowNotice'],
+    queryKey: ['useResolverShowNotifce'],
```
```suggestion
    queryKey: ['useResolverShowNotice'],
```

stackchain committed:
Update packages/resolver/src/translators/reactjs/hooks/useResolverShowNotice.tsx

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverShowNotice.tsx:
```diff
@@ -1,22 +1,24 @@
-    UseQueryOptions<boolean, Error, boolean, ['useResolverShowNotice']>
-  > = {},
+  options?: Partial<
+    UseQueryOptions<boolean, Error, boolean, ['resolver', 'show-notice']>
```
```suggestion
    UseQueryOptions<boolean, Error, boolean, ['useResolverShowNotice']>
```

stackchain committed:
Update packages/resolver/src/translators/reactjs/hooks/useResolverShowNotice.tsx

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>

Opestein committed:
Merge remote-tracking branch 'origin/Update-resolver-package' into Update-resolver-package

# Conflicts:
#	packages/resolver/src/translators/reactjs/hooks/useResolverShowNotice.tsx

stackchain commented:
@folarinemurgo can you build resolver package and test it, if it's building and there is no impact on transfer funnel we can merge it.

Opestein committed:
rn79: Update Resolver Package
Fixed TypeScript Portability Error

Opestein committed:
Revert "rn79: Update Portability Error"

This reverts commit a974014e29cb4a603e48ed5c24383f1739663f23.

Opestein committed:
rn79: Update Resolver Package
Fixed TypeScript Portability Error

folarinemurgo commented:
> @folarinemurgo can you build resolver package and test it, if it's building and there is no impact on transfer funnel we can merge it.


Pushed an update, all test cases passed now @stackchain

Opestein committed:
Merge remote-tracking branch 'origin/develop' into Update-resolver-package

# Conflicts:
#	packages/resolver/package-lock.json
#	packages/resolver/package.json

Opestein committed:
rn79: Update Resolver Package
Updated tsconfig

stackchain commented on packages/resolver/tsconfig.json:
```diff
@@ -20,10 +20,13 @@
     "paths": {
       "@tanstack/react-query": ["./node_modules/@tanstack/react-query"],
       "@tanstack/query-core": ["./node_modules/@tanstack/query-core"],
+      "react": ["./node_modules/@types/react"],
```
```suggestion
```

stackchain committed:
Update packages/resolver/tsconfig.json

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>

stackchain commented on packages/resolver/tsconfig.json:
```diff
@@ -20,6 +20,8 @@
     "skipLibCheck": true,
     "strict": true,
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
```
```suggestion
```

stackchain committed:
Update packages/resolver/tsconfig.json

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>

stackchain commented on packages/resolver/src/translators/reactjs/hooks/useResolverShowNotice.tsx:
```diff
@@ -1,22 +1,24 @@
     queryKey: ['useResolverShowNotice'],
     queryFn: showNotice.read,
+    staleTime: 0,
+    gcTime: 0,
```
```suggestion
```

stackchain committed:
Update packages/resolver/src/translators/reactjs/hooks/useResolverShowNotice.tsx

Signed-off-by: Juliano Lazzarotto <30806844+stackchain@users.noreply.github.com>
